duckduckgo/iOS
# Flame asset optimization

Task/Issue URL:
Tech Design URL:
CC:

**Description**:
Memory usage for graphics on iOS is based on the UIImage size of the stored image. The flame image was 2000x878, with 20 images in the animation. On some devices, like iPad, the full image is needed to display a high resolution animation.

In the case of a portrait iPhone 7s, or even a portrait iPhone Xs, such a large resolution is not possible to display at once. This PR better leverages the asset catalogs features for device and width specific assets to reduce memory load of the flame animation.

Specifically, assets are split into iPhone Compact Width, iPhone Any Width, iPad Compact Width, and iPad Any Width variants with associated @2x and @3x resolutions.

The reduction in run-time memory usage is significant for portrait iOS devices, especially iPhone devices loading @2x compact width assets.

Finally, app bundle size should not be impacted (I was unable to archive to test this). On some devices, due to app thinning, the bundle size might even slightly decrease.

**Steps to test this PR**:
1. Checkout PR
2. Build and run the app
3. Try the flame animation in portrait
4. Try the flame animation in landscape
5. Repeat 3-4 for supported devices
6. Archive the app using this PR and verify the estimated bundle size has not changed significantly

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
